app/models/run.rb

```diff
@@ -53,9 +57,9 @@ def program(string_or_symbol)
       h[string_or_symbol.to_s]
     end
 
-    def program_from_content_type(content_type_string)
-      Run.exportable_programs.each do |prg|
-        return prg if prg.content_type == content_type_string
+    def program_from_attribute(func_name, value)
+      Run.programs.map do |prg|
+        return prg if prg.send(func_name) == value
```

Comment:
I'm in favor of the abstraction, but `send` makes me very nervous. This has strong potential to be an attack vector if some future code passes user input in without realizing what the function is doing.

I would rather have a bit of duplicated code such as one function per `func_name`. Or, you can have this function switch/case on a specific set of `func_name` values. Either way we should not allow any arbitrary method to be called (all modules already come with [a lot of dangerous ones that we don't think about][1]).

[1]: https://ruby-doc.org/core-2.4.1/Module.html

Reply:
Is send considered one of those methods you should really never use like in JS with eval?

Reply:
yeah pretty much. i would only ever consider using it to perform magic in a library that was well-guarded against the outside world abusing it.